sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/tests/Bindings/CacheableReadBlobTests.cs

```diff
@@ -0,0 +1,192 @@
+        {
+            // Arrange
+            FunctionDataCacheKey key = new FunctionDataCacheKey("foo", "0x1");
+            bool isIncrementActiveRefs = true;
+            Mock<SharedMemoryMetadata> sharedMemMetaMock = CreateMockSharedMemoryMetadata();
+            SharedMemoryMetadata sharedMemMeta = sharedMemMetaMock.Object;
+            Mock<IFunctionDataCache> cacheMock = CreateMockFunctionDataCache();
+            cacheMock
+                .Setup(c => c.TryPut(key, sharedMemMeta, isIncrementActiveRefs, false))
+                .Returns(expected)
+                .Verifiable();
+            IFunctionDataCache cache = cacheMock.Object;
+            Mock<Stream> blobStreamMock = CreateMockBlobStream();
+            Stream blobStream = blobStreamMock.Object;
+            CacheableReadBlob cacheableReadBlob = CreateProductUnderTest(key, blobStream, cache);
+
+            // Act
+            bool result = cacheableReadBlob.TryPutToCache(sharedMemMeta, isIncrementActiveRefs);
+
+            // Assert
+            Assert.AreEqual(expected, result);
+            cacheMock.Verify();
+        }
+
+        [Test]
+        public void TryPutToCacheAlreadyCached_VerifyFailure()
+        {
+            // Arrange
+            FunctionDataCacheKey key = new FunctionDataCacheKey("foo", "0x1");
+            bool isIncrementActiveRefs = true;
+            Mock<IFunctionDataCache> cacheMock = CreateMockFunctionDataCache();
+            cacheMock
+                .Setup(c => c.TryPut(key, It.IsAny<SharedMemoryMetadata>(), isIncrementActiveRefs, false))
+                .Throws(new Exception("This should not be called"));
+            IFunctionDataCache cache = cacheMock.Object;
+            Mock<SharedMemoryMetadata> sharedMemMetaMock = CreateMockSharedMemoryMetadata();
+            SharedMemoryMetadata sharedMemMeta = sharedMemMetaMock.Object;
+            CacheableReadBlob cacheableReadBlob = CreateProductUnderTest(key, sharedMemMeta, cache);
+
+            // Act
+            bool result = cacheableReadBlob.TryPutToCache(sharedMemMeta, isIncrementActiveRefs);
+
+            // Assert
+            Assert.AreEqual(false, result);
+        }
+
+        [Test]
+        public void CacheHit_Dispose_VerifyCacheRefCountDecremented()
+        {
+            // Arrange
+            FunctionDataCacheKey key = new FunctionDataCacheKey("foo", "0x1");
+            Mock<SharedMemoryMetadata> sharedMemMetaMock = CreateMockSharedMemoryMetadata();
+            SharedMemoryMetadata sharedMemMeta = sharedMemMetaMock.Object;
+            Mock<IFunctionDataCache> cacheMock = CreateMockFunctionDataCache();
+            cacheMock
+                .Setup(c => c.DecrementActiveReference(key))
+                .Verifiable();
+            IFunctionDataCache cache = cacheMock.Object;
+            CacheableReadBlob cacheableReadBlob = CreateProductUnderTest(key, sharedMemMeta, cache);
+
+            // Act
+            cacheableReadBlob.Dispose();
+
+            // Assert
```

Comment:
You use it before but here might be a good place to explain what the cache mock verifies as an assert